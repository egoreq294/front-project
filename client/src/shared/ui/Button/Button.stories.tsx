import type { Meta, StoryObj } from '@storybook/react';

import { Button } from './Button';

const meta: Meta<typeof Button> = {
  title: 'shared/Button',
  component: Button,
  argTypes: {},
};

export default meta;
type Story = StoryObj<typeof Button>;

export const Primary: Story = {
  args: {
    children: 'Button',
  },
};

export const Ghost: Story = {
  args: {
    variant: 'Ghost',
    children: 'Button',
  },
};

export const Filled: Story = {
  args: {
    variant: 'Filled',
    children: 'Button',
  },
};

export const Success: Story = {
  args: {
    variant: 'Success',
    children: 'Button',
  },
};

export const Danger: Story = {
  args: {
    variant: 'Danger',
    children: 'Button',
  },
};

export const PrimaryDisabled: Story = {
  args: {
    children: 'Button',
    disabled: true,
  },
};
