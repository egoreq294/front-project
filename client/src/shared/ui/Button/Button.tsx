import cn from 'classnames';
import React, { ButtonHTMLAttributes, FC, ReactNode } from 'react';

import styles from './styles.module.scss';

type ButtonVariant = 'Primary' | 'Ghost' | 'Filled' | 'Danger' | 'Success';

type Size = 'S' | 'M' | 'L';

interface ButtonProps extends ButtonHTMLAttributes<HTMLButtonElement> {
  className?: string;
  children: ReactNode;
  size?: Size;
  variant?: ButtonVariant;
  disabled?: boolean;
  fullWidth?: boolean;
  testId?: string;
}

export const Button: FC<ButtonProps> = ({
  className,
  variant = 'Primary',
  size = 'S',
  children,
  disabled,
  fullWidth,
  testId,
  ...props
}) => (
  <button
    type="button"
    data-testid={testId ? `Button_${testId}` : undefined}
    className={cn(
      styles.Button,
      styles[variant],
      styles[size],
      { [styles.FullWidth]: fullWidth },
      className,
    )}
    disabled={disabled}
    {...props}
  >
    {children}
  </button>
);
