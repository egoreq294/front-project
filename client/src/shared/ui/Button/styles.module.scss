.Button {
  cursor: pointer;
  color: var(--color-text);
  background: none;
  outline: none;
  border: none;
  display: flex;
  gap: var(--gutter-xxs);
  align-items: center;
  justify-content: center;

  &:disabled {
    color: var(--color-hint);
    cursor: not-allowed;
  }

  &:hover:not(:disabled) {
    color: var(--color-accent);
  }
}

.S {
  padding: var(--gutter-xxxs) var(--gutter-xs);
  font: var(--new-font-s);
}

.M {
  padding: var(--gutter-xs) var(--gutter-s);
  font: var(--new-font-m);
}

.L {
  padding: var(--gutter-s) var(--gutter-m);
  font: var(--new-font-l);
}

.Primary {
  border: 1px solid var(--color-icon);
  border-radius: var(--border-radius);
}

.Filled {
  background: var(--color-background-light);
  border-radius: var(--border-radius);
}

.Danger {
  border: 1px solid var(--color-cancel);
  border-radius: var(--border-radius);
}

.Success {
  border: 1px solid var(--color-save);
  border-radius: var(--border-radius);
}

.FullWidth {
  width: 100%;
}
