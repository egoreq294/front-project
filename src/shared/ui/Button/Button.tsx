import React, { ButtonHTMLAttributes, FC, ReactNode } from 'react';
import cn from 'classnames';
import styles from './styles.module.scss';

type ButtonVariant =
  | 'Primary'
  | 'PrimaryInverted'
  | 'Secondary'
  | 'SecondaryInverted'
  | 'Ghost'
  | 'GhostInverted';

type Size = 'S' | 'M' | 'L';

interface ButtonProps extends ButtonHTMLAttributes<HTMLButtonElement> {
  className?: string;
  children: ReactNode;
  size?: Size;
  variant?: ButtonVariant;
}

export const Button: FC<ButtonProps> = ({
  className,
  variant = 'Primary',
  size = 'S',
  children,
  ...props
}) => (
  <button
    type="button"
    className={cn(styles.Button, styles[variant], styles[size], className)}
    {...props}
  >
    {children}
  </button>
);
