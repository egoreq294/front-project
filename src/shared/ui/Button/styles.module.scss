.Button {
    cursor: pointer;
    color: var(--inverted-primary-color);
}

.Clear {
    padding: 0;
    margin: 0;
    border: 0;
    background: none;
    outline: none;
}