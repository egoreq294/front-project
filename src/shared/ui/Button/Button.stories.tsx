import type { Meta, StoryObj } from '@storybook/react';

import { Button } from './Button';

const meta: Meta<typeof Button> = {
  title: 'shared/Button',
  component: Button,
  argTypes: {},
};

export default meta;
type Story = StoryObj<typeof Button>;

export const Primary: Story = {
  args: {
    children: 'Button',
  },
};

export const Ghost: Story = {
  args: {
    variant: 'Ghost',
    children: 'Button',
  },
};

export const PrimaryDisabled: Story = {
  args: {
    children: 'Button',
    disabled: true,
  },
};

export const GhostDisabled: Story = {
  args: {
    variant: 'Ghost',
    children: 'Button',
    disabled: true,
  },
};
