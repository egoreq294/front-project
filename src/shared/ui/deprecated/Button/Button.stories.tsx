import type { Meta, StoryObj } from '@storybook/react';

import { Button } from './Button';

const meta: Meta<typeof Button> = {
  title: 'shared/deprecated/Button',
  component: Button,
  argTypes: {},
};

export default meta;
type Story = StoryObj<typeof Button>;

export const Primary: Story = {
  args: {
    children: 'Button',
  },
};

export const PrimaryInverted: Story = {
  args: {
    children: 'Button',
    variant: 'PrimaryInverted',
  },
};

export const Secondary: Story = {
  args: {
    variant: 'Secondary',
    children: 'Button',
  },
};

export const SecondaryInverted: Story = {
  args: {
    variant: 'SecondaryInverted',
    children: 'Button',
  },
};

export const Ghost: Story = {
  args: {
    variant: 'Ghost',
    children: 'Button',
  },
};

export const GhostInverted: Story = {
  args: {
    variant: 'GhostInverted',
    children: 'Button',
  },
};

export const Danger: Story = {
  args: {
    variant: 'Danger',
    children: 'Button',
  },
};

export const PrimaryDisabled: Story = {
  args: {
    children: 'Button',
    disabled: true,
  },
};
