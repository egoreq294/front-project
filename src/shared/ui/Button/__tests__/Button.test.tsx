import { render, screen } from '@testing-library/react';
import { Button, ThemeButton } from '../Button';
import React from 'react';

describe('Button test suite', () => {
  test('should render correctly', () => {
    const { container } = render(<Button>TEST</Button>);
    expect(container).toMatchSnapshot();
  });
  test('should render correctly with TEST children', () => {
    render(<Button>TEST</Button>);
    expect(screen.getByText('TEST')).toBeInTheDocument();
  });
  test('should render correctly with clear theme', () => {
    render(<Button theme={ThemeButton.CLEAR}>TEST</Button>);
    expect(screen.getByText('TEST')).toHaveClass('Clear');
  });
});
