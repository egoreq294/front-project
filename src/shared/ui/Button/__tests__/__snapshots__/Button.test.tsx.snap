// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button test suite should render correctly 1`] = `
<div>
  <button
    class="Button undefined"
    type="button"
  >
    TEST
  </button>
</div>
`;
